web/source/kmwosk.ts

```diff
@@ -292,18 +295,16 @@ if(!window['keyman']['initialized']) {
      * @return      {Object.<string,number>}   Array object with position and size of OSK container
      * Description  Get rectangle containing KMW Virtual Keyboard
      */
-    osk['getRect'] = osk.getRect = function()			// I2405
-    {
-      var p={};
-      if(osk._DivVKbd)
-      {
+    osk['getRect'] = osk.getRect = function() {		// I2405
+      var p: any;
```

Comment:
Surely `var p: Object`?

Reply:
Defining `var p: Object` causes the `p.left`, `p.top`, etc assignments to be errors.  (Not `p['left']` or `p['top']`, though.)  TypeScript doesn't allow non-associative-array style assignments to the type `Object` because they are not predefined properties of the `Object` type.  TypeScript doesn't allow declaration of new members of a class that were not previously part of its definition, unless declared as part of an extended interface for the class.  (See kmwexthtml.ts for examples, but we shouldn't extend the base class of the overall type hierarchy, I'd think.)

Either we define a proper type with the desired properties or we use `any` to allow these assignments to proceed unhindered.

Reply:
True. So long term I guess your plan is to tighten this up as well with a new type declaration (interface or class).

Reply:
Yes.  Granted, it's looking like kmwosk.ts will be the last file to be tightened up.